src/transport_interface/transport_interface_test.c

```diff
@@ -421,26 +522,137 @@ static void prvSendRecvCompareFunc( void * pParam )
     for( testBufferSize = 1; testBufferSize <= TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH; testBufferSize = testBufferSize + 1 )
     {
         /* Initialize the test data buffer. */
-        prvInitializeTestData( pTrasnportTestBufferStart, testBufferSize );
+        prvInitializeTestData( pTransportTestBufferStart, testBufferSize );
 
         /* Send the test data to the server. */
         if( pThreadParameter->xResult == true )
         {
-            pThreadParameter->xResult = prvTransportSendData( pTestTransport, pNetworkContext, pTrasnportTestBufferStart,
+            pThreadParameter->xResult = prvTransportSendData( pTestTransport,
+                                                              pNetworkContext,
+                                                              pTransportTestBufferStart,
+                                                              testBufferSize );
+        }
+
+        /* Receive the test data from server. */
+        if( pThreadParameter->xResult == true )
+        {
+            pThreadParameter->xResult = prvTransportRecvData( pTestTransport,
+                                                              pNetworkContext,
+                                                              pTransportTestBufferStart,
                                                               testBufferSize );
         }
 
+        /* Compare the test data received from server. */
+        if( pThreadParameter->xResult == true )
+        {
+            pThreadParameter->xResult = prvVerifyTestData( pTransportTestBufferStart,
+                                                           testBufferSize,
+                                                           TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH );
+        }
+
+        if( ( pThreadParameter->stopFlag == true ) || ( pThreadParameter->xResult == false ) )
+        {
+            break;
+        }
+
+        #if ( TRANSPORT_TEST_PRINT_DEBUG_PROGRESS == 1 )
+            /* Output information to indicate the test is running. */
+            UNITY_OUTPUT_CHAR( '.' );
+        #endif
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+#ifdef TRANSPORT_TEST_EXECUTE_WRITEV_TESTS
+/**
+ * @brief Thread function of the writev/receive/compare test.
+ */
+static void prvWritevRecvCompareFunc( void * pParam )
+{
+    threadParameter_t * pThreadParameter = pParam;
+    uint8_t * pTransportTestBufferStart =
+        &( pThreadParameter->transportTestBuffer[ TRANSPORT_TEST_BUFFER_PREFIX_GUARD_LENGTH ] );
+    uint32_t testBufferSize = 0;

```

Comment:
There's a mix of constants in this function that are given signs with "U" and some that aren't. I think it might make sense to be consistent and either have them all be signed or have them all be unsigned, given that testBufferSize is a uint32_t